### ISSUE TITLE
no longer getting completing reads for compilation and other run commands

### ISSUE THREAD

**Issue by rmmanseau** [bbatsov/projectile#1844]
## Expected behavior
interactively running projectile-compile-project or projectile-run-project
should open a completing read so that I can filter & select previously run commands

## Actual behavior
I get a prompt. I can use arrow keys to scroll up and get previous commands, but I don't see them all in the mini buffer.

## Steps to reproduce the problem

I switched from ivy to vertico recently and it stopped working for compilation. but I don't think its related to that, because I'm getting completing-reads for other projectile commands, ie find file in project, file recent files, etc. Im wondering if there has been a change in behavior. Digging thru the projectile source it doesn't seem like `projectile-compile-project` or `projectile-run-project` make use of the completing read function anymore

### Projectile version information

```
Projectile version: 2.7.0
```

### Emacs version

`28.2` running doom emacs

### Operating system

Mac OS

**Comment by bbatsov** [comment]
I don't think we ever had this type of completion for compilation commands. Perhaps ivy was treating this specially in some way? I haven't used it in a while and I can't remember at this point.

**Comment by bbatsov** [comment]
Projectile reads these commands with `read-shell-command`, which gives you shell history (`M-p`/`M-n`) but not a completion menu - and that's always been the case. The behavior you remember was almost certainly counsel-projectile/ivy treating the command history specially, not Projectile's own code, so nothing here regressed.

Closing. A proper completing-read over previous commands would be a reasonable separate feature request if someone wants to build it.
